latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+        self._python_inputs = inputs
+        self._python_outputs = outputs
+        self._target_file_for_input_param = target_file_for_input_param
+        self._target_file_for_output_param = target_file_for_output_param
+
+        def placeholder():
+            ...
+
+        self._task_function = placeholder
+
+        super().__init__(
+            task_type=task_type,
+            name=name,
+            interface=interface,
+            task_config=None,
+            task_resolver=SnakemakeJobTaskResolver(),
+        )
+
+    def get_fn_code(self):
+
+        code_block = ""
+
+        fn_interface = f"\n\n@small_task\ndef {self.name}("
+        for idx, (param, t) in enumerate(self._python_inputs.items()):
+            fn_interface += f"{param}: {t.__name__}"
+            if idx == len(self._python_inputs) - 1:
+                fn_interface += ")"
+            else:
+                fn_interface += ", "
+
+        if len(self._python_outputs.items()) > 0:
+            for idx, (param, t) in enumerate(self._python_outputs.items()):
+                if idx == 0:
+                    fn_interface += f" -> NamedTuple('{self.name}_output', "
+                fn_interface += f"{param}={t.__name__}"
+                if idx == len(self._python_outputs) - 1:
+                    fn_interface += "):"
+                else:
+                    fn_interface += ", "
+        else:
+            fn_interface += ":"
+
+        code_block += fn_interface
+
+        for param, t in self._python_inputs.items():
+            if t == LatchFile:
+                code_block += f'\n\tPath({param}).resolve().rename(ensure_parents_exist(Path("{self._target_file_for_input_param[param]}")))'
+
+        snakemake_cmd = ["snakemake"]
+        snakemake_cmd.extend(
+            ["--target-jobs", *encode_target_jobs_cli_args(self.job.get_target_spec())]
+        )
+        snakemake_cmd.extend(["--allowed-rules", *self.job.rules])
+        snakemake_cmd.extend(["--local-groupid", str(self.job.jobid)])
+        snakemake_cmd.extend(["--cores", str(self.job.threads)])
+
+        if not self.job.is_group():
+            snakemake_cmd.append("--force-use-threads")
+
+        excluded = {"_nodes", "_cores", "tmpdir"}
+        allowed_resources = list(
+            filter(lambda x: x[0] not in excluded, self.job.resources.items())
+        )
+        if len(allowed_resources) > 0:
+            snakemake_cmd.append("--resources")
+            for resource, value in allowed_resources:
+                snakemake_cmd.append(f"{resource}={value}")
+
+        formatted_snakemake_cmd = "\n\n\tsubprocess.run(["
+
+        for i, arg in enumerate(snakemake_cmd):
+            arg_wo_quotes = arg.strip('"').strip("'")
+            formatted_snakemake_cmd += f'"{arg_wo_quotes}"'
+            if i == len(snakemake_cmd) - 1:
+                formatted_snakemake_cmd += "], check=True)"
+            else:
+                formatted_snakemake_cmd += ", "
```

Comment:
```suggestion
        formatted_snakemake_cmd = "\n\n\tsubprocess.run([" + repr(snakemake_cmd) + "], check=True)"
```
why wouldn't this just work